oauth-service-impl/src/main/java/de/adorsys/psd2/oauth/service/impl/TokenServiceImpl.java

```diff
@@ -104,37 +114,43 @@ public TokenBO save(TokenBO token) {
     }
 
     @Override
-    public void delete(String id) {
-        logger.info("Deleting token by id={}", id);
-        repository.delete(id);
-    }
+    public String attachState(String scaOAuthLink, String aspspId, String psuId) {
+        if (isEmpty(scaOAuthLink) || isEmpty(aspspId)) {
+            throw new IllegalArgumentException("scaOAuth link and aspspId must be not empty");
+        }
+        if (isEmpty(psuId)) {
+            psuId = UUID.randomUUID().toString();
+        }
 
-    private static class State {
-        private String clientId;
-        private String aspspId;
+        OauthStateBO state = new OauthStateBO(psuId, aspspId);
+        String encodedState = encodeState(state);
 
-        public String getClientId() {
-            return clientId;
-        }
+        Matcher matcher = STATE_PATTERN.matcher(scaOAuthLink);
 
-        public void setClientId(String clientId) {
-            this.clientId = clientId;
+        if (matcher.find()) {
+            StringBuffer sb = new StringBuffer();
+            String quote = Pattern.quote(matcher.group(1));
+            matcher.appendReplacement(sb, matcher.group(0).replaceFirst(quote, encodedState));
+            matcher.appendTail(sb);
```

Comment:
Could you, please, extract lines 131-134 (or even 135 as well) into some private method with a meaningful name (probably it will return String) to improve this piece of code readability?

Reply:
done